base/src/main/java/org/mozilla/jss/netscape/security/x509/Extensions.java

```diff
@@ -40,7 +40,7 @@
  * @version 1.11
  * @see CertAttrSet
  */
-public class Extensions extends Vector<Extension>
+public class Extensions extends ArrayList<Extension>
```

Comment:
I'd suggest we complete the `synchronized` changes in this PR first, then do the `Vector` replacement in a separate PR because that's possibly going to be much more complicated.

I'm actually hoping we could replace `Vector` variables/fields with `ArrayList` first. Replacing `Vector` in class inheritance can be done too, but as @ckelleyRH pointed above, it's likely to break backward compatibility. If we're OK with that, I think we should actually drop it from the class inheritance and make it a field instead, e.g.:

```
public class Extensions {
    ArrayList<Extension> extensions = new ArrayList<>();
}
```

This way other applications will be isolated from future changes in the internal JSS implementation. This is something that we can discuss in the new PR.

Reply:
OK, I will create a new commit with only the few `synchronized` methods which make sense to fix now and move the discussion to new PRs which I would consider for all the collections.